### DISCUSSION TITLE
Is there any Notification to application when tendermint panic and stop?

### DISCUSSION THREAD

**Discussion by lcmmhcc** [tendermint/tendermint#9748] [Q&A] [answered]
for example,  CONSENSUS FAILURE accurred or other panic.  the application may need to finalize or do something.

**Comment by thanethomson** [comment, marked as answer]
For built-in applications (like applications based on the Cosmos SDK), the panic will naturally propagate through the entire process.

For socket-based applications, the application will lose its network connection to Tendermint, allowing it more freedom in choosing what exactly to do.
